x/gamm/keeper/grpc_query_test.go

```diff
@@ -4,11 +4,151 @@ import (
 	gocontext "context"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/types/query"
 
 	"github.com/osmosis-labs/osmosis/v12/x/gamm/types"
 )
 
+func (suite *KeeperTestSuite) TestCalcExitPoolCoinsFromShares() {
+	queryClient := suite.queryClient
+	ctx := suite.Ctx
+	poolId := suite.PrepareBalancerPool()
+	exitFee := sdk.ZeroDec()
+
+	testCases := []struct {
+		name          string
+		poolId        uint64
+		tokenOut      string
+		shareInAmount sdk.Int
+		expectedErr   error
+	}{
+		{
+			"valid test case",
+			poolId,
+			"foo",
```

Comment:
please aim to use `sdk.DefaultBondDenom` or a reusable consant instead of hardcoding the denom name. The reason is it is more difficult to maintain if we keep hard coding more things

Reply:
unfortunately suite.prepareBalancerPool does not include stake denom, this is the reason I did not use it. If you agree, I can change this method to creatte a pool with stake denom

Reply:
Makes sense. Let's not block this then